sharedbuffers/mapped_struct.py

```diff
@@ -1377,7 +1382,7 @@ def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0):
         See `mapped_object.pack_into` for argument details.
         """
         cur_offs = offs
-        if six.PY3:
+        if python3:
             objlen = obj.nbytes
```

Comment:
I'm not sure about using .nbytes just like that.

Why don't you take a memoryview explicitly to do the cast to bytes?

We want to be polymorphic here and accept byte strings, bytearrays, and iterables of int. Using nbytes straightaway breaks for byte strings and bytearrays both.

In fact, maybe add a test to make sure this works with byte stirngs and byte arrays.

Reply:
> I'm not sure about using .nbytes just like that.

ok, I changed this

> We want to be polymorphic here and accept byte strings, bytearrays, and iterables of int. Using nbytes straightaway breaks for byte strings and bytearrays both.

`proxied_buffer.unpack_from` returns a `buffer`. So the `self.assertEquals(obj, unpacked_obj)` fails if `obj` is a string (bytes) in py2

Reply:
Maybe the test can cast the expected object to buffer in the assert then:

``` python
self.assertEquals(buffer(obj), unpacked_obj)
```